schema.c

```diff
@@ -1257,6 +1293,32 @@ sch_node_child (sch_node * parent, const char *child)
     return _sch_node_child (ns, parent, child);
 }
 
+sch_node *
+sch_node_ns_child (xmlNs *ns, sch_node * parent, const char *child)
+{
+    return _sch_node_child (ns, parent, child);
+}
+
+sch_node *
+sch_node_match_schema_name (sch_node * schema, const char *target_name)
```

Comment:
is this trying to find a sibling with a particular name? rather than sch_node_child which finds a child with a name?

Reply:
Yes, that is what it is doing.

Reply:
ok .. in that case a better prototype would be
sch_node *sch_node_child (sch_node *parent, const char *name);
sch_node *sch_node_child_first (sch_node * parent);
sch_node *sch_node_sibling (sch_node * node, const char *name);
sch_node *sch_node_next_sibling (sch_node * node);

but you could also just use sch_node_next_sibling to find what you are looking for

Reply:
I have used sch_node_next_sibling and removed sch_node_match_schema_name.